matlpotlib 3.10.0 test failures?

I don't know how long https://github.com/DougBurke/sherpa/actions/runs/12601910769/job/35123965032 will remain, but this shows failures with matplotlib 3.10.0 (released mid December 2024) 

Looks like a couple of tests need tweaks; this doesn't appear to be a fundamental issue

```
=================================== FAILURES ===================================
_________________________ test_pha1_plot_data_options __________________________

caplog = <_pytest.logging.LogCaptureFixture object at 0x13f0f4fd0>
clean_astro_ui = None, requires_pylab = None, basic_pha1 = None

    @requires_fits
    @requires_data
    def test_pha1_plot_data_options(caplog, clean_astro_ui, requires_pylab,
                                    basic_pha1):
        """Test that the options have changed things, where easy to do so"""
    
        from matplotlib import pyplot as plt
        import matplotlib
    
        prefs = ui.get_data_plot_prefs('tst')
    
        # check the preference are as expected for the boolean cases
        assert not prefs['xerrorbars']
        assert prefs['yerrorbars']
        assert not prefs['xlog']
        assert not prefs['ylog']
    
        prefs['xerrorbars'] = True
        prefs['yerrorbars'] = False
        prefs['xlog'] = True
        prefs['ylog'] = True
    
        prefs['color'] = 'orange'
    
        # linecolor is unused, but check we can set it as existing code
        # may reference it.
        prefs['linecolor'] = 'brown'
        prefs['linestyle'] = '-.'
    
        prefs['marker'] = 's'
        prefs['markerfacecolor'] = 'cyan'
        prefs['markersize'] = 10
    
        with caplog.at_level(logging.INFO, logger='sherpa'):
            ui.plot_data()
    
        # check for linecolor warning
        assert len(caplog.record_tuples) == 1
        rec = caplog.record_tuples[0]
        assert len(rec) == 3
        loc, lvl, msg = rec
    
        assert loc == 'sherpa.plot.pylab_backend'
        assert lvl == logging.WARNING
        assert msg == 'The linecolor attribute (brown) is unused.'
    
        ax = plt.gca()
        assert ax.get_xscale() == 'log'
        assert ax.get_yscale() == 'log'
    
        assert ax.get_xlabel() == 'Energy (keV)'
        assert ax.get_ylabel() == 'Counts/sec/keV'
    
        # It is not clear whether an 'exact' check on the value, as
        # provided by pytest.approx, makes sense, or whether a "softer"
        # check - e.g.  just check whether it is less- or greater- than a
        # value - should be used. It depends on how often matplotlib
        #
        line = ax.lines[1]
        assert line.get_color() != 'green'  # option over-ridden
        assert line.get_linestyle() == 'None'  # option over-ridden
        assert line.get_marker() == 's'
        assert line.get_markerfacecolor() == 'cyan'  # line.get_color()  # option over-ridden
        assert line.get_markersize() == pytest.approx(10.0)
        assert line.get_alpha() == pytest.approx(0.7)
    
        # MODEL - points
        #
        pts = ax.lines[2]
        assert pts.get_color() != 'green'  # option over-ridden
        assert pts.get_linestyle() == '-'
        assert pts.get_marker() == 'None'
        assert pts.get_markerfacecolor() == '#1f77b4'  # line.get_color()  # option over-ridden
        assert pts.get_markersize() == pytest.approx(6.0)
        assert pts.get_alpha() == pytest.approx(0.7)
    
        # assume error bars handled by a collection; test a subset
        # of values
        #
        assert len(ax.collections) == 1
        coll = ax.collections[0]
    
        assert len(coll.get_segments()) == 42
    
        # The return value depends on matplotlib version (>= 3.3
        # returns something). What has changed? Maybe this should
        # not be tested?
        #
        expected = [(None, None)]
        if matplotlib.__version__ >= '3.3.0':
            expected = [(0.0, None)]
    
>       assert coll.get_linestyles() == expected
E       assert [(np.float64(0.0), None)] == [(None, None)]
E         
E         At index 0 diff: (np.float64(0.0), None) != (None, None)
E         
E         Full diff:
E           [
E               (
E         -         None,
E         +         np.float64(0.0),
E                   None,
E               ),
E           ]

sherpa/astro/ui/tests/test_astro_ui_plot.py:2052: AssertionError
---------------------------- Captured stdout setup -----------------------------
WARNING: systematic errors were not found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.pi'
statistical errors were found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.pi'
but not used; to use them, re-read with use_errors=True
read ARF file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.arf
read RMF file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.rmf
WARNING: systematic errors were not found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi'
statistical errors were found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi'
but not used; to use them, re-read with use_errors=True
read background file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi
dataset tst: 0.00146:14.9504 -> 0.4672:9.8696 Energy (keV)
------------------------------ Captured log setup ------------------------------
WARNING  sherpa.astro.io:__init__.py:817 systematic errors were not found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.pi'
INFO     sherpa.astro.io:__init__.py:824 statistical errors were found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.pi'
but not used; to use them, re-read with use_errors=True
INFO     sherpa.astro.io:__init__.py:703 read ARF file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.arf
INFO     sherpa.astro.io:__init__.py:703 read RMF file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273.rmf
WARNING  sherpa.astro.io:__init__.py:817 systematic errors were not found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi'
INFO     sherpa.astro.io:__init__.py:824 statistical errors were found in file '/Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi'
but not used; to use them, re-read with use_errors=True
INFO     sherpa.astro.io:__init__.py:924 read background file /Users/runner/work/sherpa/sherpa/sherpa/../sherpa-test-data/sherpatest/3c273_bg.pi
INFO     sherpa.ui.utils:utils.py:270 dataset tst: 0.00146:14.9504 -> 0.4672:9.8696 Energy (keV)
--------------------------- Captured stdout teardown ---------------------------
 Solar Abundance Vector set to angr:  Anders E. & Grevesse N. Geochimica et Cosmochimica Acta 53, 197 (1989)
 Cross Section Table set to vern:  Verner, Ferland, Korista, and Yakovlev 1996
=============================== warnings summary ===============================
```

--- a/sherpa/astro/ui/tests/test_astro_ui_plot.py
+++ b/sherpa/astro/ui/tests/test_astro_ui_plot.py
@@ -1,5 +1,5 @@
 #
-#  Copyright (C) 2019 - 2024
+#  Copyright (C) 2019 - 2025
 #  Smithsonian Astrophysical Observatory
 #
 #
@@ -32,6 +32,7 @@ correct data; it is more a check that the routines can be called.
 
 import copy
 import logging
+
 import numpy as np
 
 import pytest
@@ -1843,15 +1844,15 @@ def test_pha1_plot_data_options(caplog, clean_astro_ui, requires_pylab,
 
     assert len(coll.get_segments()) == 42
 
-    # The return value depends on matplotlib version (>= 3.3
-    # returns something). What has changed? Maybe this should
-    # not be tested?
+    # The linestyles return value has changed with matplotib
+    # and NumPy version, so just check the second element
+    # and not the first (which can be None, 0.0, np.float64(0)).
     #
-    expected = [(None, None)]
-    if matplotlib.__version__ >= '3.3.0':
-        expected = [(0.0, None)]
-
-    assert coll.get_linestyles() == expected
+    #
+    ls = coll.get_linestyles()
+    assert len(ls) == 1
+    assert len(ls[0]) == 2
+    assert ls[0][1] is None
 
     # looks like the color has been converted to individual channels
     # - e.g. floating-point values for R, G, B, and alpha.
@@ -2041,15 +2042,15 @@ def test_pha1_plot_fit_options(clean_astro_ui, requires_pylab, basic_pha1):
 
     assert len(coll.get_segments()) == 42
 
-    # The return value depends on matplotlib version (>= 3.3
-    # returns something). What has changed? Maybe this should
-    # not be tested?
+    # The linestyles return value has changed with matplotib
+    # and NumPy version, so just check the second element
+    # and not the first (which can be None, 0.0, np.float64(0)).
     #
-    expected = [(None, None)]
-    if matplotlib.__version__ >= '3.3.0':
-        expected = [(0.0, None)]
-
-    assert coll.get_linestyles() == expected
+    #
+    ls = coll.get_linestyles()
+    assert len(ls) == 1
+    assert len(ls[0]) == 2
+    assert ls[0][1] is None
 
     # looks like the color has been converted to individual channels
     # - e.g. floating-point values for R, G, B, and alpha.

